tools/chplcheck/src/rules.py

```diff
@@ -95,42 +101,88 @@ def MethodsAfterFields(context, node):
                 method_seen = True
         return True
 
+    #Five things have to match between consecutive decls for this to warn:
+    # 1. same type
+    # 2. same kind
+    # 3. same attributes
+    # 4. same linkage
+    # 5. same pragmas
     @driver.advanced_rule(default=False)
     def ConsecutiveDecls(context, root):
         def is_relevant_decl(node):
+            var_node = None
             if isinstance(node, MultiDecl):
+                
                 for child in node:
-                    if isinstance(child, Variable): return child.kind()
+                    if isinstance(child, Variable): var_node = child
             elif isinstance(node, Variable):
-                return node.kind()
-            return None
-
-        def recurse(node, skip_direct = False):
+                var_node = node
+            
+            if var_node is None:
+                return None
+            
+            var_type = None
+            var_type_expr = var_node.type_expression()
+
+            if isinstance(var_type_expr, FnCall):
+                #for function call, we need to match all the components
+                var_type = ''
+                for child in var_type_expr:
+                    if child is None:
+                        continue
+                    if 'name' in dir(child):
+                        var_type += child.name()
+                    elif 'text' in dir(child):
+                        var_type += child.text()
+            elif isinstance(var_type_expr, Identifier):
+                var_type = var_type_expr.name()
+            
+            var_kind = var_node.kind()
+
+            var_attributes = ''
+            var_attribute_group = var_node.attribute_group()
+            if var_attribute_group is not None:
+                for attribute_node in var_attribute_group:
+                    if attribute_node is not None:
+                        var_attributes += attribute_node.name() + " "
```

Comment:
Python style nit. What you have if fine and readable, I think this is more pythonic and more readable 

```
if var_attribute_group := var_node.attribute_group():
  var_attributes = " ".join([a.name() for a in var_attribute_group if a is not None])
```